xds/internal/balancer/edsbalancer/eds_impl.go

```diff
@@ -402,7 +406,7 @@ func (ebwcc *edsBalancerWrapperCC) UpdateBalancerState(state connectivity.State,
 }
 
 func (ebwcc *edsBalancerWrapperCC) UpdateState(state balancer.State) {
-	ebwcc.parent.updateState(ebwcc.priority, state)
+	ebwcc.parent.enqueueChildBalancerStateUpdate(ebwcc.priority, state)
```

Comment:
Instead of calling into the `edsBalancer` through this hook and making use of it's `run` goroutine to break the deadlock, what if we just spun off a goroutine here:

```golang
go func() {
  ebwcc.parent.updateState(ebwcc.priority, state)
}()
```
It seems a little excessive to me to go through the `edsBalancer` and back to `edsImpl`. But maybe I am missing something.

And once we split the priority balancer out, we will anyways have it's own `run` goroutine.

Reply:
I just started looking further into the PR and what I have suggested is exactly what you have done in the `TestEDS_ChildPolicyUpdatePickerInline`.

Reply:
We cannot call update in a new goroutine every time, because the order of the of the update matters. A early goroutine may be scheduled to run after a newer one, so the older balancer state will override the newer state.
We have to add the states into a queue, and process them in order.

Reply:
Makes sense. Thanks.